### PR TITLE
[database] bump version to 31 adding second columns to some indices

### PR DESCRIPTION
**Pull request by HansBull**
After analyzing a couple of queries I think there is still much potential in improving the use of indices. This is a first attempt. See also https://discuss.pixls.us/t/slow-image-import-to-a-large-library/22327

### CODE REVIEW

**Comment by johnny-bit** [comment]
Good!

One thing is missing though: please add those indices to `_create_library_schema` too otherwise new users won't be able to take advantage of them.

**Comment by HansBull** [comment]
Thanks. Missed that.

File: src/common/database.c [hunk 1]
```diff
@@ -1675,6 +1675,43 @@ static int _upgrade_library_schema_step(dt_database_t *db, int version)
...
+        "[init] can't drop images_film_id_index\n");
+    TRY_EXEC("CREATE INDEX `images_film_id_index` ON `images` ( `film_id`, `filename` )",
+        "[init] can't recreate images_film_id_index\n");
+
+    TRY_EXEC("DROP INDEX IF EXISTS `images_group_id_index`",
+        "[init] can't drop images_group_id_index\n");
+    TRY_EXEC("CREATE INDEX `images_group_id_index` ON `images` ( `group_id`, `id` )",
+        "[init] can't recreate images_group_id_index\n");
+
+    TRY_EXEC("DROP INDEX IF EXISTS `masks_history_imgid_index`",
```
Comment (TurboGit) [inline src/common/database.c:1703]: it is changed ok, but new database will still use the previous definition, you need to update the definition in `_create_library_schema()`. Likewise for others indexes.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Can you share the perfs improvement measured ? TIA.

**Comment by TurboGit** [comment]
> One thing is missing though: please add those indices to `_create_library_schema` too otherwise new users won't be able to take advantage of them.

I did the same comment without seeing yours first :)

File: src/common/database.c (hunk 1 above)
Comment (HansBull) [reply src/common/database.c:1703]: Was changed in https://github.com/darktable-org/darktable/pull/7747/commits/7675ee2158f1564c535a646f9e25f003434de3db

**Comment by HansBull** [comment]
> Can you share the perfs improvement measured ? TIA.

Well that's kind of a problem. One measurement of one index change was done by fatman on a large collection here https://discuss.pixls.us/t/slow-image-import-to-a-large-library/22327/13
Unfortunately I'm on a secondary machine without a large collection (the main machine being in another country due to the virus), so I really can't respond.

The `images_latlong_index` should benefit (maybe @phweyland could give it a try) with regard to the query in map.c line 2393  (`SELECT id, longitude, latitude FROM %s WHERE longitude >= ?1 AND longitude <= ?2 AND latitude <= ?3 AND latitude >= ?4 AND longitude NOT NULL AND latitude NOT NULL)` since without the index a `EXPLAIN QUERY PLAN` run with that very query results in a full table scan, while with the index it gives `SEARCH TABLE images USING COVERING INDEX images_latlong_index (latitude>? AND latitude<?)`.

**Comment by TurboGit** [comment]
Ok, I have read the discussion in pixls forum, seems a nice speed-up. I'll test on my side.

**Review by TurboGit — APPROVED** [review]
Ok, let's move forward, all good on my side. Thanks.

**Pull request merged**